libcaf_core/caf/uri.cpp

```diff
@@ -249,8 +254,12 @@ std::string to_string(const uri& x) {
 
 std::string to_string(const uri::authority_type& x) {
   std::string str;
-  if (!x.userinfo.empty()) {
-    uri::encode(str, x.userinfo);
+  if (x.userinfo) {
+    uri::encode(str, x.userinfo->name);
+    if (x.userinfo->password) {
+      str += ':';
+      uri::encode(str, *x.userinfo->password);
```

Comment:
Can be less repetitive:

```suggestion
    if (auto& pw = x.userinfo->password) {
      str += ':';
      uri::encode(str, *pw);
```